src/Effects.php

```diff
@@ -0,0 +1,252 @@
+    /**
+     * @param \DOMDocument $document
+     */
+    public function __construct(\DOMDocument $document)
+    {
+        $this->document = $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function gamma($correction)
+    {
+        $gamma = (float) $correction;
+
+        if ($gamma <= 0) {
+            throw new InvalidArgumentException(sprintf(
+                'Invalid gamma correction value %s, must be a positive float or interger',
+                var_export($correction, true)
+            ));
+        }
+
+        $exponent = 1 / $gamma ** 2;
+
+        $this->addFilterElement('feComponentTransfer', [
+            ['feFuncR', [
+                'type' => 'gamma',
+                'exponent' => $exponent,
+            ]],
+            ['feFuncG', [
+                'type' => 'gamma',
+                'exponent' => $exponent,
+            ]],
+            ['feFuncB', [
+                'type' => 'gamma',
+                'exponent' => $exponent,
+            ]],
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function negative()
+    {
+        $this->addFilterElement('feColorMatrix', [
+            'type' => 'matrix',
+            'values' => implode(' ', [
+                '-1 0 0 0 1',
+                '0 -1 0 0 1',
+                '0 0 -1 0 1',
+                '0 0  0 1 0',
+            ]),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function grayscale()
+    {
+        $this->addFilterElement('feColorMatrix', [
+            'type' => 'saturate',
+            'values' => '0',
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function colorize(ColorInterface $color)
+    {
+        if (!$color instanceof RGB) {
+            throw new NotSupportedException('Colorize with non-rgb color is not supported');
+        }
+
+        $this->addFilterElement('feColorMatrix', [
+            'type' => 'matrix',
+            'values' => implode(' ', [
+                '1 0 0 0 '.json_encode($color->getRed() / 255),
+                '0 1 0 0 '.json_encode($color->getGreen() / 255),
+                '0 0 1 0 '.json_encode($color->getBlue() / 255),
+                '0 0 0 1 0',
+            ]),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sharpen()
+    {
+        $this->addFilterElement('feConvolveMatrix', [
+            'kernelMatrix' => implode(' ', [
+                '-0.02 -0.12 -0.02',
+                '-0.12  1.56 -0.12',
+                '-0.02 -0.12 -0.02',
+            ]),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function blur($sigma)
+    {
+        $deviation = (float) $sigma;
+
+        if ($deviation <= 0) {
+            throw new InvalidArgumentException(sprintf(
+                'Invalid sigma %s, must be a positive float or interger',
```

Comment:
Typo: inte**r**ger

Reply:
Fixed in 72a3839e44ac2db353bb945aad736961b9dce158